core/index/src/main/java/mil/nga/giat/geowave/core/index/CompoundIndexStrategy.java

```diff
@@ -0,0 +1,377 @@
+		defaultNumberOfRanges = (int) Math.ceil(Math.pow(
+				2,
+				getNumberOfDimensions()));
+	}
+
+	/**
+	 * Get the number of dimensions of each sub-strategy
+	 *
+	 * @return an array with the number of dimensions for each sub-strategy
+	 */
+	public int[] getNumberOfDimensionsPerIndexStrategy() {
+		return new int[] {
+			subStrategy1.getOrderedDimensionDefinitions().length,
+			subStrategy2.getOrderedDimensionDefinitions().length
+		};
+	}
+
+	/**
+	 * Get the total number of dimensions from all sub-strategies
+	 *
+	 * @return the number of dimensions
+	 */
+	public int getNumberOfDimensions() {
+		return baseDefinitions.length;
+	}
+
+	/**
+	 * Create a compound ByteArrayId
+	 *
+	 * @param id1
+	 *            ByteArrayId for the first sub-strategy
+	 * @param id2
+	 *            ByteArrayId for the second sub-strategy
+	 * @return the ByteArrayId for the compound strategy
+	 */
+	public ByteArrayId composeByteArrayId(
+			final ByteArrayId id1,
+			final ByteArrayId id2 ) {
+		final byte[] bytes = new byte[id1.getBytes().length + id2.getBytes().length + 4];
+		final ByteBuffer buf = ByteBuffer.wrap(bytes);
+		buf.put(id1.getBytes());
+		buf.put(id2.getBytes());
+		buf.putInt(id1.getBytes().length);
+		return new ByteArrayId(
+				bytes);
+	}
+
+	/**
+	 * Get the ByteArrayId for each sub-strategy from the ByteArrayId for the
+	 * compound index strategy
+	 *
+	 * @param id
+	 *            the compound ByteArrayId
+	 * @return the ByteArrayId for each sub-strategy
+	 */
+	public ByteArrayId[] decomposeByteArrayId(
+			final ByteArrayId id ) {
+		final ByteBuffer buf = ByteBuffer.wrap(id.getBytes());
+		final int id1Length = buf.getInt(id.getBytes().length - 4);
+		final byte[] bytes1 = new byte[id1Length];
+		final byte[] bytes2 = new byte[id.getBytes().length - id1Length - 4];
+		buf.get(bytes1);
+		buf.get(bytes2);
+		return new ByteArrayId[] {
+			new ByteArrayId(
+					bytes1),
+			new ByteArrayId(
+					bytes2)
+		};
+	}
+
+	private List<ByteArrayId> composeByteArrayIds(
+			final List<ByteArrayId> ids1,
+			final List<ByteArrayId> ids2 ) {
+		final List<ByteArrayId> ids = new ArrayList<>(
+				ids1.size() * ids2.size());
+		for (final ByteArrayId id1 : ids1) {
+			for (final ByteArrayId id2 : ids2) {
+				ids.add(composeByteArrayId(
+						id1,
+						id2));
+			}
+		}
+		return ids;
+	}
+
+	private ByteArrayRange composeByteArrayRange(
+			final ByteArrayRange rangeOfStrategy1,
+			final ByteArrayRange rangeOfStrategy2 ) {
+		final ByteArrayId start = composeByteArrayId(
+				rangeOfStrategy1.getStart(),
+				rangeOfStrategy2.getStart());
+		final ByteArrayId end = composeByteArrayId(
+				rangeOfStrategy1.getEnd(),
+				rangeOfStrategy2.getEnd());
+		return new ByteArrayRange(
+				start,
+				end);
+	}
+
+	private List<ByteArrayRange> getByteArrayRanges(
+			final List<ByteArrayRange> ranges1,
+			final List<ByteArrayRange> ranges2 ) {
+		final List<ByteArrayRange> ranges = new ArrayList<>(
+				ranges1.size() + ranges2.size());

```

Comment:
not a big deal because the list is going to dynamically resize anyways, but if you want to set a good initial size shouldn't it be ranges1.size() * ranges2.size()?